### PR TITLE
Adding post destroy checks for vSphere platform

### PR DESCRIPTION
**Pull request by vavuthu**
Signed-off-by: vavuthu <vavuthu@redhat.com>

### CODE REVIEW

File: ocs_ci/deployment/vmware.py [hunk 1]
```diff
@@ -233,6 +233,23 @@ def modify_scaleup_repo(self):
...
+        Post destroy checks on cluster
+
+        Args:
+            pool (str): Resource pool name
+            dc (str): Datacenter name
+            cluster (str): Cluster name
+
+        """
+        if self.vsphere.is_resource_pool_exist(pool, dc, cluster):
+            logger.warning(f"Resource pool {pool} exists even after "
```
Comment (petr-balogh) [inline ocs_ci/deployment/vmware.py]:
```
logger.warning(
    f"msg"
)
```

File: ocs_ci/deployment/vmware.py [hunk 2]
```diff
@@ -233,6 +233,23 @@ def modify_scaleup_repo(self):
         # change root disk size
         change_vm_root_disk_size(constants.SCALEUP_VSPHERE_MACHINE_CONF)
 
+    def post_destroy_checks(self, pool, dc, cluster):
```
Comment (vasukulkarni) [inline ocs_ci/deployment/vmware.py]: Could be done for pre check as well ? This can help cleanup existing pool if its somehow missed by manual cleanup.

File: ocs_ci/utility/vsphere.py [hunk 3]
```diff
@@ -384,3 +384,86 @@ def restart_vms(self, vms, force=True):
...
+            dc (str): Datacenter name
+            cluster (str): Cluster name
+
+        Returns:
+            bool: True if resource pool exists, otherwise False
+
+        """
+        return True if self.get_pool(pool, dc, cluster) else False
+
+    def poweroff_vms(self, vms):
```
Comment (vasukulkarni) [inline ocs_ci/utility/vsphere.py]: You should actually move this to different PR as it seems unrelated to this fix.

File: ocs_ci/utility/vsphere.py [hunk 4]
```diff
@@ -384,3 +384,86 @@ def restart_vms(self, vms, force=True):
...
+        for vm in vms:
+            status = self.get_vm_power_status(vm)
+            logger.info(f"power state of {vm.name}: {status}")
+            if status == "poweredOn":
+                to_poweroff_vms.append(vm)
+        logger.info(f"Powering off VMs: {[vm.name for vm in to_poweroff_vms]}")
+        tasks = [vm.PowerOff() for vm in to_poweroff_vms]
+        WaitForTasks(tasks, self._si)
+
+    def poweron_vms(self, vms):
```
Comment (vasukulkarni) [inline ocs_ci/utility/vsphere.py]: same

File: ocs_ci/utility/vsphere.py (hunk 3 above)
Comment (clacroix12) [reply ocs_ci/utility/vsphere.py]: `poweroff_vms` is called by `destroy_vms` which is called by `destroy_pool` which is called by `post_destroy_checks ` :)

Comment (vasukulkarni) [reply ocs_ci/utility/vsphere.py]: thanks :)

File: ocs_ci/deployment/vmware.py (hunk 1 above)
Comment (vavuthu) [reply ocs_ci/deployment/vmware.py]: Done

File: ocs_ci/deployment/vmware.py (hunk 2 above)
Comment (vavuthu) [reply ocs_ci/deployment/vmware.py]: problem with pre-check is if someone has same cluster name, it will delete the cluster
